components/eamxx/src/dynamics/homme/eamxx_homme_iop.cpp

```diff
@@ -144,7 +156,12 @@ advance_iop_subsidence(const KT::MemberType& team,
     Pack delta_tracer_k, delta_tracer_km1;
     for (int iq=0; iq<n_q_tracers; ++iq) {
       auto s_delta_tracer = Kokkos::subview(s_delta_Q, iq, Kokkos::ALL());
-      ekat::index_and_shift<-1>(s_delta_tracer, range_pack_m1, delta_tracer_k, delta_tracer_km1);
+      ekat::index_and_shift<-1>(s_delta_tracer, range_pack_for_m1_shift, delta_tracer_k, delta_tracer_km1);
+      if (at_top.any() or at_bot.any()) {
```

Comment:
Here too, I would break the if into two, to avoid running the `set` method of the one you don't have (since we never have both of them true at the same time).

Reply:
Do you think it makes sense to cache the value of `at_top.any()` in a local var? It may be a micro (or nano) opt, so probably not really needed? Although, for MAM4xx cases, we may have close to 100 tracers...so maybe precomputing `const bool at_top_any = at_top.any();` and similarly for `at_bot`, may prove helpful?